src/huggingface_hub/hub_mixin.py

```diff
@@ -0,0 +1,231 @@
+            name = model_id
+        else:
+            _, name = model_id.split("/")
+
+        if name in os.listdir() and CONFIG_NAME in os.listdir(name):
+            print("LOADING weights from local directory")
+            config_file = os.path.join(name, CONFIG_NAME)
+        else:
+            try:
+                config_url = hf_hub_url(model_id, filename=CONFIG_NAME, revision=revision)
+                config_file = cached_download(config_url, cache_dir=cache_dir, force_download=force_download, proxies=proxies, resume_download=resume_download, local_files_only=local_files_only, use_auth_token=use_auth_token)
+            except:
+                config_file = None
+
+        if name in os.listdir():
+            print("LOADING weights from local directory")
+            model_file = os.path.join(name, PYTORCH_WEIGHTS_NAME)
+        else:
+            model_url = hf_hub_url(model_id, filename=PYTORCH_WEIGHTS_NAME, revision=revision)
+            model_file = cached_download(model_url, cache_dir=cache_dir, force_download=force_download, proxies=proxies, resume_download=resume_download, local_files_only=local_files_only, use_auth_token=use_auth_token)
+
+        if config_file is not None:
+            with open(config_file, "r", encoding="utf-8") as f:
+                config = json.load(f)
+            model = cls(config, **kwargs)
+        else:
+            model = cls(**kwargs)
+
+        state_dict = torch.load(model_file, map_location=map_location)
+        model.load_state_dict(state_dict, strict=strict)
+        model.eval()
+
+        return model
+
+    def _init_git(self, model_id, save_directory, branch):
+
+        os.chdir(save_directory)
+        if not os.path.isdir(".git"):
+            subprocess.run(["git", "init"], stdout=subprocess.PIPE)
+
+        if not os.path.isdir(".git/lfs"):
+            subprocess.run(["git-lfs", "install"], stdout=subprocess.PIPE)
+
+        if branch not in os.listdir(".git/refs/heads"):
+            subprocess.run(["git", "checkout", "-b", branch], stdout=subprocess.PIPE)
+
+        with open(".git/HEAD") as f:
+            content = f.read().split("/")
+        if content[-1][:-1] != branch:
+            subprocess.run(["git", "checkout", branch], stdout=subprocess.PIPE)
+
+        with open(".git/config", "r") as f:
+            git_config = f.read().split()
+        if (PREFIX+model_id) not in git_config:
+            subprocess.run(["git", "remote", "add", "origin", PREFIX+model_id], stdout=subprocess.PIPE)
+            subprocess.run(["git", "fetch", "origin"], stdout=subprocess.PIPE)
+            if branch in os.listdir(".git/refs/remotes/origin"):
+                subprocess.run(["git", "merge", f"origin/{branch}"], stdout=subprocess.PIPE)
+        os.chdir("../")
+
+    def _push_git(self, save_directory, track_extensions, commit_message, branch):
+
+        os.chdir(save_directory)
+        subprocess.run(["git-lfs", "track"]+track_extensions, stdout=subprocess.PIPE)
```

Comment:
If user wants to add file corresponding to some other extension lets say `.pt`, then it will allow them to do that. Currently `.gitattributes` is not having this extension which is very common.  Also, if user wants to commit to some branch `ver1` (different from `main`) without relating commit histories of 2 branches, then its is possible only if he add `.gitattributes` manually by this command.

The reason, I am fetching and merging (instead of cloning repo) is that, it will give freedom to name local directory whatever they want (will help if they doing epoch pushing during training) and still keeping the hub repo name to `model_id`.